src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -481,8 +477,31 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                                                     INS_OPTS_NONE);
                     }
                 }
-            }
-            break;
+                break;
+            case NI_Vector64_Create:
+            case NI_Vector128_Create:
+                if (intrin.op1->isContainedFltOrDblImmed())
+                {
+                    const double dataValue = intrin.op1->AsDblCon()->gtDconVal;
+                    GetEmitter()->emitIns_R_F(INS_fmov, emitSize, targetReg, dataValue, opt);
```

Comment:
I would add an `assert(opt != INS_OPTS_NONE)` to make sure that we are using `C7.2.122 FMOV (vector, immediate)` and not `C7.2.125 FMOV (scalar, immediate)`